apps/sms/test/marionette/mocks/mock_navigator_moz_mobile_message.js

```diff
@@ -103,6 +103,48 @@ Services.obs.addObserver(function(document) {
     });
   }
 
+  function getMessageNumbers(message) {
+    var numbers = [];
+
+    if (message.sender) {
+      numbers.push(message.sender);
+    }
+
+    if (message.receiver) {
+      numbers.push(message.receiver);
+    }
+
+    if (message.receivers) {
+      for(var i = 0; i < message.receivers.length; i++) {
+        numbers.push(message.receivers[i]);
+      }
+    }
+
+    return numbers;
+  }
+
+  function getMessagesFromFilter(threads, filter) {
+    var messages = [];
+
+    if (filter.threadId) {
+      messages = threads.get(filter.threadId).messages;
+    } else if (filter.numbers && filter.numbers.length) {
+      for (var thread of threads.values()) {
+        for(var message of thread.messages) {
```

Comment:
Mmm yes I can do if you want :) I'll need additional ```filteredMessages``` variable to do ```messages.push(...filteredMessages)``` since several filter results should be put into single array